### DISCUSSION TITLE
Uploader java.io.FileNotFoundException: my_ns_url/api/v1/entries

### DISCUSSION THREAD

**Discussion by AgushkaNightscout** [j-kaltes/Juggluco#126] [Q&A] [answered]
About 24 hours after switching on the Uploader, the data stops being sent to the Nightscout, after restarting the Juggluco transmits unsent data and then stops transmitting again. An error appears in the Uploader menu. API secret is correct, Nightscout variable auth_default_roles=status-only, Nightscout 15.0.2. Please tell me what can be the problem?

**Comment by j-kaltes** [comment]
I don't think that export AUTH_DEFAULT_ROLES=status-only or export auth_default_roles=status-only matters.

When you copy and past my_ns_url to the internet browser on your smartphone, does it work?

**Reply by AgushkaNightscout** [reply]
I've been having this problem for 4 days now and the value was "denied" before. I changed it to "status-only" and everything worked, but after about 24 hours it broke again and I thought it was important to specify. 

Here is the journalctl of my Nightscout.service. 1 - I accessed the site from my phone's browser, 2 - went data from Juggluco right after it restarted. 3 - another batch of data. After that, no data is going till now.
![Снимок экрана от 2024-02-20 18-21-27](https://github.com/j-kaltes/Juggluco/assets/90107639/c2acb8f2-9a95-4f22-bb21-d2b73996635e)

**Reply by j-kaltes** [reply]
You show only three minutes. What if you grep through your logs for a longer period:
data loaded: 
```bash
grep -aE 'data loaded|tick'  outputfile.txt
```
Where outputfile.txt is a file to which you redirected the output of your server.

**Reply by AgushkaNightscout** [reply]
Here's archive of data from 14th February till now. Please, look.
[feb14_now.txt.tar.gz](https://github.com/j-kaltes/Juggluco/files/14349643/feb14_now.txt.tar.gz)

**Reply by j-kaltes** [reply]
Every minute there is a "data loaded:"  entry. Strange enough the Nightscout server doesn't log errors differently from failure, so it says little.
I can recreate an error message like your title, when I add '/' to the Nightscout url, using https://night.myhost.net:7898/ instead of https://night.myhost.net:7898
Maybe you left out an slash in this title.
Using as Nightscout URL the url of another web server also generates this error message. It is possible that two servers use the same network port. The one started first will be using that port. But Nightscout will then exit with something like the following error message: `Error: listen EADDRINUSE: address already in use :::3333`
Changes in the modem port redirection can also make that another web server is used.
Every minute a "data loaded:" entry, speaks against this possibility, but maybe that is somehow from OpenAPS.
I can't find anything to change in the Nightscout environmental variables to generates your error message. This are the variables I have set in my start script:
```bash
#!/bin/bash
mkdir /tmp/public

#export AUTH_DEFAULT_ROLES=denied
export DISPLAY_UNITS="mmol/L"
export MONGO_CONNECTION="mongodb://name:pass@127.0.0.1:27017/Nightscout"
export BASE_URL="https://hostname:3333"
export PORT=3333
export API_SECRET="mysecret"
export PUMP_FIELDS="reservoir battery status"
export DEVICESTATUS_ADVANCED=true
#export API3_SECURITY_ENABLE=true
export API3_MAX_LIMIT=4000
export ENABLE="careportal loop iob cob openaps pump bwg rawbg basal cors direction timeago devicestatus ar2 profile boluscalc food sage iage cage alexa basalprofile bgi directions bage upbat googlehome errorcodes reservoir battery openapsbasal"
export TIME_FORMAT=24
export INSECURE_USE_HTTP=true
export SECURE_HSTS_HEADER=false
export LANGUAGE=en
export EDIT_MODE=on
export PUMP_ENABLE_ALERTS=true
export PUMP_FIELDS="reservoir battery clock status"
export PUMP_RETRO_FIELDS="reservoir battery clock"
export PUMP_WARN_CLOCK=30
export PUMP_URGENT_CLOCK=60
export PUMP_WARN_RES=50
export PUMP_URGENT_RES=10
export PUMP_WARN_BATT_P=30
export PUMP_URGENT_BATT_P=20
export PUMP_WARN_BATT_V=1.35
export PUMP_URGENT_BATT_V=1.30
export OPENAPS_ENABLE_ALERTS=false
export OPENAPS_WARN=30
export OPENAPS_URGENT=60
export OPENAPS_FIELDS="status-symbol status-label iob meal-assist rssi freq"
export OPENAPS_RETRO_FIELDS="status-symbol status-label iob meal-assist rssi"
export LOOP_ENABLE_ALERTS=false
export LOOP_WARN=30
export LOOP_URGENT=60
export SHOW_PLUGINS=careportal
export SHOW_FORECAST="ar2 openaps"

export SSL_KEY=privkey.pem
export SSL_CERT=fullchain.pem
export SSL_CA=fullchain.pem
# start server
node server.js
```

**Reply by AgushkaNightscout** [reply, marked as answer]
I've been testing this problem for a few days. Everything is ok with my Nightscout variables, I rechecked several times. At first I thought the problem was in the nginx settings, I even found error 431 in logs and increased client_max_body_size to 20M, but it didn't help. According to nginx logs and tcpdump of opened  inсoming port of my vps, the data came exactly when I rebooted Juggluco and then did not come. Then I checked that the data from the phone was leaving by installing PCAPdroid - it also recorded the data transfer when I restarted Juggluco and nothing else. As a result of experiments I came to the following conclusion - the data is only transferred after a Juggluco reboot, if the phone with the Uploader is a wi-fi hotspot. If the phone is attached to the wi-fi or does not switch on the hotspot on it, the data is transferred all the time and without problems. Apparently, the problem is in the hardware. BUT! When I updated to version 7.5.1 today, the problem disappeared)))) Now I can share my LTE internet by switching on the hotspot and data from Juggluco to Nightscout comes in normally.
